test/test_helper_script.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from re import UNICODE
+
+import pytest
+
+from cve_bin_tool.error_handler import UnknownArchiveType
+from cve_bin_tool.helper_script import HelperScript
+
+
+class TestHelperScript:
+    @pytest.mark.parametrize(
+        "filename, product_name",
+        [
+            (
+                "libsndfile-1.0.25-12.el7.x86_64.rpm",
+                "libsndfile",
+            ),  # normal packages (with 2 '-')
+            (
+                "libsndfile-1.0.31-1-aarch64.pkg.tar.xz",
+                "libsndfile",
+            ),  # .tar.xz packages (with 3 '-')
+            (
+                "logrotate_3.14.0-4ubuntu5_amd64.deb",
+                "logrotate",
+            ),  # .deb packages (with 2 '_')
+            (
+                "libarchive_3.4.2-1_aarch64_cortex-a72.ipk",
+                "libarchive",
+            ),  # .ipk packages (with 3 '_')
+            (
+                "tomcat9_9.0.37-3_all.deb",
+                "tomcat9",
+            ),  # packages with digits in product_name
+            (
+                "openssh-client_8.4p1-5ubuntu1_amd64.deb",
+                "openssh-client",
+            ),  # packages with '-' in product_name
+            (
+                "./Packages/bash-4.2.46-34.el7.x86_64.rpm",
+                "bash",
+            ),  # directaory names
+        ],
+    )
+    def test_parse_filename(self, filename, product_name):
+        hs = HelperScript(filename)
+        assert hs.parse_filename(filename) == product_name
+
+    # @pytest.mark.parametrize("filename", [
+    #     "bash-4.2.46-34.el7.x86_64.abc" # unsupported file type
+    # ])
+    # def test_parse_filename(self, filename):
+    #     hs = HelperScript(filename)
+    #     with pytest.raises(UnknownArchiveType):
+    #         hs.parse_filename(filename)
```

Comment:
@anthonyharrison I am raising `UnknownArchiveType` error in `helper_script.py`, but this does not seem to pass. the error message in pytest was - 

```
(venv39_CVE_Binary_tool) peb@LAPTOP-M375SCBF:/mnt/d/git_stuff/cve-bin-tool$ pytest test/test_helper_script.py
======================================================== test session starts ========================================================
platform linux -- Python 3.8.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /mnt/d/git_stuff/cve-bin-tool
plugins: forked-1.3.0, asyncio-0.15.1, cov-2.12.0, xdist-2.2.1
collected 1 item

test/test_helper_script.py F                                                                                                  [100%]

============================================================= FAILURES ==============================================================
________________________________ TestHelperScript.test_parse_filename[bash-4.2.46-34.el7.x86_64.abc] ________________________________

self = <cve_bin_tool.helper_script.HelperScript object at 0x7f475e2e46a0>, filename = 'bash-4.2.46-34.el7.x86_64.abc'

    def parse_filename(self, filename):
        """
        returns package_name/product_name from package_filename of types .rpm, .deb, etc.
        Example: package_filename = openssh-client_8.4p1-5ubuntu1_amd64.deb
            here, package_name = openssh-client
        """

        filename = filename.split("/")[-1]  # for directory names
        if self.extractor.can_extract(filename):
            if filename.endswith(".tar.xz"):
                product_name = filename.rsplit("-", 3)[0]
                # example: libarchive-3.5.1-1-aarch64.pkg.tar.xz
            elif filename.endswith(".deb") or filename.endswith(".ipk"):
                product_name = filename.rsplit("_")[0]
            else:
                product_name = filename.rsplit("-", 2)[0]
            self.logger.debug(f"The product name is {product_name}")

            return product_name
        else:
            # self.logger.error("Unknown Archive Type")
            with ErrorHandler(mode=ErrorMode.NoTrace, logger=self.logger):
>               raise UnknownArchiveType(filename)
E               cve_bin_tool.error_handler.UnknownArchiveType: bash-4.2.46-34.el7.x86_64.abc

cve_bin_tool/helper_script.py:55: UnknownArchiveType

During handling of the above exception, another exception occurred:

self = <test.test_helper_script.TestHelperScript object at 0x7f475e2e46d0>, filename = 'bash-4.2.46-34.el7.x86_64.abc'

    @pytest.mark.parametrize("filename", [
        "bash-4.2.46-34.el7.x86_64.abc" # unsupported file type
    ])
    def test_parse_filename(self, filename):
>       hs = HelperScript(filename)

test/test_helper_script.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cve_bin_tool/helper_script.py:20: in __init__
    self.product = product_name or self.parse_filename(filename)
cve_bin_tool/helper_script.py:55: in parse_filename
    raise UnknownArchiveType(filename)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cve_bin_tool.error_handler.ErrorHandler object at 0x7f475e2e4640>
exc_type = <class 'cve_bin_tool.error_handler.UnknownArchiveType'>, exc_val = UnknownArchiveType('bash-4.2.46-34.el7.x86_64.abc')
exc_tb = <traceback object at 0x7f475e2f8400>

    def __exit__(self, exc_type, exc_val, exc_tb):
        if isinstance(exc_val, BaseException):
            self.exit_code = ERROR_CODES.get(exc_type, -1)
            self.exc_val = exc_val
        if self.mode == ErrorMode.Ignore:
            return True
        if exc_type:
            if self.logger and exc_val:
                self.logger.error(f"{exc_type.__name__}: {exc_val}")
            if self.mode == ErrorMode.NoTrace:
>               sys.exit(self.exit_code)
E               SystemExit: -13

cve_bin_tool/error_handler.py:144: SystemExit
--------------------------------------------------------- Captured log call ---------------------------------------------------------
ERROR    cve_bin_tool.HelperScript:error_handler.py:142 UnknownArchiveType: bash-4.2.46-34.el7.x86_64.abc
====================================================== short test summary info ======================================================
FAILED test/test_helper_script.py::TestHelperScript::test_parse_filename[bash-4.2.46-34.el7.x86_64.abc] - SystemExit: -13
========================================================= 1 failed in 0.46s =========================================================
```

Reply:
Try going through this https://github.com/intel/cve-bin-tool/blob/fce0bc88d825fdd74f1df72fd7dabd3e93f1500c/test/test_package_list_parser.py#L44

I think you need to add the exception in parametrize

Reply:
@BreadGenie tried this, but no luck :(
does this has something to with me not having `error_mode` in `Helper_script`
```
class HelperScript:
    """Helps Contributors to understand how various common filenames, version-strings, etc. look"""

    def __init__(self, filename, vendor_name=None, product_name=None):
```

Reply:
could be
I had error mode in the `PackageListParser` class